python/lsst/ts/dream/csc/dream_csc.py

```diff
@@ -132,43 +166,128 @@ async def begin_disable(self, id_data: salobj.BaseDdsDataType) -> None:
-        if self.disabled_or_enabled:
-            if not self.connected:
-                await self.connect()
-        else:
-            await self.disconnect()
+        self.weather_loop_task.cancel()
+        await self.weather_loop_task
+
+        if self.ess_remote is not None:
+            await self.ess_remote.close()
+            self.ess_remote = None
+
+        self.log.info("Disconnected.")
 
     async def configure(self, config: SimpleNamespace) -> None:
         self.config = config
 
     @property
     def connected(self) -> bool:
-        return not self.model.connected
+        if self.model is None:
+            return False
+        return self.model.connected
 
     @staticmethod
     def get_config_pkg() -> str:
         return "ts_config_ocs"
 
     async def do_pause(self, data: salobj.BaseMsgType) -> None:
-        # To be implemented later.
-        pass
+        await self.disconnect()
 
     async def do_resume(self, data: salobj.BaseMsgType) -> None:
-        # To be implemented later.
-        pass
+        if not self.connected:
+            await self.connect()
+
+    async def weather_loop(self) -> None:
+        """Periodically check weather station, and send a flag if needed.
+
+        A weather flag is sent when the weather has changed, or every ten
+        minutes.
+        """
+        if not self.config:
+            raise RuntimeError("Not yet configured")
+
+        while self.model is not None and self.model.connected:
+            if self.simulation_mode == 0:
+                if self.ess_remote is None:
+                    self.ess_remote = salobj.Remote(
+                        domain=self.domain, name="ESS", index=self.config.ess_index
+                    )
+                    await self.ess_remote.start_task
+                    self.log.debug(f"Connected to ESS:{self.config.ess_index}.")
+
+                    self.weather_ok_flag = None
+                    self.last_weather_update = 0.0
+                    if self.ess_remote is None:
+                        self.log.error("Failed to connect to weather CSC.")
+                        continue
+
+                try:
+                    # Get weather data.
+                    weather_ok_flag = True
+                    air_flow = await self.ess_remote.tel_airFlow.next(
+                        flush=False, timeout=2
+                    )
+                    if air_flow is None or air_flow.speed > 25:
+                        weather_ok_flag = False
+
+                    precipitation = await self.ess_remote.evt_precipitation.get()
+                    if precipitation is None or (
+                        precipitation.raining or precipitation.snowing
+                    ):
+                        weather_ok_flag = False
+                except Exception:
+                    self.log.exception("Failed to read weather data from ESS.")
+                    self.ess_remote = None
```

Comment:
Here you need to call `await self.ess_remote.close()` as well before setting to None.